python-checks/src/main/java/org/sonar/python/checks/RobustCipherAlgorithmCheck.java

```diff
@@ -63,14 +86,54 @@ public class RobustCipherAlgorithmCheck extends PythonSubscriptionCheck {
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, subscriptionContext -> {
       CallExpression callExpr = (CallExpression) subscriptionContext.syntaxNode();
-      Symbol calleeSymbol = callExpr.calleeSymbol();
-      if (calleeSymbol != null) {
-        String fqn = calleeSymbol.fullyQualifiedName();
-        if (fqn != null && sensitiveCalleeFqns.contains(fqn)) {
-          subscriptionContext.addIssue(callExpr.callee(), MESSAGE);
-        }
-      }
+      Optional.ofNullable(callExpr)
+        .map(CallExpression::calleeSymbol)
+        .map(Symbol::fullyQualifiedName)
+        .filter(fqn -> sensitiveCalleeFqns.contains(fqn) || SSL_SET_CIPHERS_FQN.equals(fqn))
+        .ifPresent(str -> addMethodSpecificIssue(subscriptionContext, callExpr, str));
     });
   }
 
+  private static void addMethodSpecificIssue(
+    SubscriptionContext subscriptionContext,
+    CallExpression callExpression,
+    String fullyQualifiedName
+  ) {
+    if (!SSL_SET_CIPHERS_FQN.equals(fullyQualifiedName) || argumentSpecifiesRiskyAlgorithm(callExpression)) {
+      subscriptionContext.addIssue(callExpression.callee(), MESSAGE);
+    }
+  }
+
+  private static boolean argumentSpecifiesRiskyAlgorithm(CallExpression callExpression) {
+    return Optional.of(callExpression.arguments())
+      .filter(list -> list.size() == 1)
+      .map(list -> list.get(0))
+      .flatMap(TreeUtils.toOptionalInstanceOfMapper(RegularArgument.class))
+      .map(RegularArgument::expression)
+      .map(RobustCipherAlgorithmCheck::unpackArgument)
+      .filter(RobustCipherAlgorithmCheck::containsInsecureCipher)
+      .isPresent();
+  }
+
+  @CheckForNull
+  private static String unpackArgument(@CheckForNull Expression expression) {
```

Comment:
I think in the argument list the annotation should be @Nullable.

Reply:
Updated in a18d2c0663b4df943eb27a458edc0490c7d7ee48.